pallets/offchain-demo/src/lib.rs

```diff
@@ -171,35 +202,72 @@ impl<T: Trait> Module<T> {
 		}
 	}
 
-	fn fetch_n_parse(_block_number: T::BlockNumber) -> Result<(), Error<T>> {
+	/// Check if we have fetched github info before. If yes, we use the cached version that is
+	///   stored in off-chain worker storage `storage`. If no, we fetch the remote info and then
+	///   write the info into the storage for future retrieval.
+	fn fetch_if_needed() -> Result<(), Error<T>> {
+
+		// Start off by creating a reference to Local Storage value.
+		// Since the local storage is common for all offchain workers, it's a good practice
+		// to prepend our entry with the pallet name.
+		let storage = StorageValueRef::persistent(b"offchain-demo::gh-info");
+
+		// The local storage is persisted and shared between runs of the offchain workers,
+		// and offchain workers may run concurrently. We can use the `mutate` function, to
+		// write a storage entry in an atomic fashion.
+		//
+		// It has a similar API as `StorageValue` that offer `get`, `set`, `mutate`.
+		// If we are using a get-check-set access pattern, we likely want to use `mutate` to access
+		// the storage in one go.
+		//
+		// Ref: https://substrate.dev/rustdocs/v2.0.0-alpha.6/sp_runtime/offchain/storage/struct.StorageValueRef.html
+		let res = storage.mutate(|store: Option<Option<GithubInfo>>| {
+			match store {
+				// info existed, returning the value
+				Some(Some(info)) => {
+					debug::info!("Using cached gh-info.");
+					Ok(info)
+				},
+				// info not existed, so we remote fetch (and parse the JSON)
+				_ => Self::fetch_n_parse(),
+			}
+		});
+
+		// The value of `res` looks funny. Its type is `Result<Result<T, E>, E>`. The above
```

Comment:
Since you are not using all the values this probably means that you don't really need `mutate` API at all and could use a simpler pattern here.